### PR TITLE
[WIP] pdb: use capman.suspend in setup

### PR DESCRIPTION
**Pull request by blueyed**
Followup to d406786a.

TODO:

- [x] test
- [x] changelog

### CODE REVIEW

File: testing/test_pdb.py [hunk 1]
```diff
@@ -1177,3 +1177,62 @@ def runcall(self, *args, **kwds):
...
+    child.expect(r'-> print\("print_in_f"\)')
+    child.expect(r"\(Pdb")
+    child.sendline("c")
+    rest = child.read().decode("utf8")
+    assert "Captured stdout call" in rest
+    assert "print_in_f" in rest
+    assert "1 failed" in rest
+
+
+def test_continue_with_breakpoint_capsys(testdir):
```
Comment (nicoddemus) [inline testing/test_pdb.py]: Here as well:

```
(.env36) λ git log -1
commit 7a121d30495ddf4304be6961fdd0a76b57fc5d88 (HEAD -> blueyed/pdb-setup-suspend)
Author: Daniel Hahler <git@thequod.de>
Date:   Thu Apr 11 11:32:09 2019 +0200

    Add test_continue_with_breakpoint_capsys


(.env36) λ cat .tmp\test-pdb.py
def f():
    print("print_in_f")
    assert 0


def test_1(capsys):
    __import__('pdb').set_trace()
    f()


(.env36) λ pytest .tmp\test-pdb.py
======================================= test session starts =======================================
platform win32 -- Python 3.6.6, pytest-4.4.1.dev67+g7a121d304, py-1.6.0, pluggy-0.9.0
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('c:\\pytest\\.hypothesis\\examples')
rootdir: c:\pytest\.tmp, inifile: pytest.ini
plugins: hypothesis-3.74.2
collected 1 item

.tmp\test-pdb.py
>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> c:\pytest\.tmp\test-pdb.py(8)test_1()
-> f()
(Pdb) break f
Breakpoint 1 at c:\pytest\.tmp\test-pdb.py:1
(Pdb) continue

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB continue (IO-capturing resumed) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> c:\pytest\.tmp\test-pdb.py(2)f()
-> print("print_in_f")
(Pdb) c

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB continue (IO-capturing resumed) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
F                                                                           [100%]

============================================ FAILURES =============================================
_____________________________________________ test_1 ______________________________________________

capsys = <_pytest.capture.CaptureFixture object at 0x000001D30D046A90>

    def test_1(capsys):
        __import__('pdb').set_trace()
>       f()

.tmp\test-pdb.py:8:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    def f():
        print("print_in_f")
>       assert 0
E       assert 0

.tmp\test-pdb.py:3: AssertionError
-------------------------------------- Captured stdout call ---------------------------------------
print_in_f
==================================== 1 failed in 29.38 seconds ====================================
Exception ignored in: <async_generator object _ag at 0x000001D30B3570B0>
Traceback (most recent call last):
  File "C:\Users\Bruno\AppData\Local\Programs\Python\Python36\lib\types.py", line 27, in _ag
  File "C:\Users\Bruno\AppData\Local\Programs\Python\Python36\lib\bdb.py", line 53, in trace_dispatch
  File "C:\Users\Bruno\AppData\Local\Programs\Python\Python36\lib\bdb.py", line 79, in dispatch_call
  File "C:\Users\Bruno\AppData\Local\Programs\Python\Python36\lib\bdb.py", line 176, in break_anywhere
  File "C:\Users\Bruno\AppData\Local\Programs\Python\Python36\lib\bdb.py", line 36, in canonic
AttributeError: 'NoneType' object has no attribute 'abspath'
```

File: testing/test_pdb.py [hunk 2]
```diff
@@ -1177,3 +1177,62 @@ def runcall(self, *args, **kwds):
     )
     assert result.ret == 0
     result.stdout.fnmatch_lines(["*runcall_called*", "* 1 passed in *"])
+
+
+def test_continue_with_breakpoint(testdir):
+    """Test capturing with later breakpoint."""
```
Comment (nicoddemus) [inline testing/test_pdb.py]: Taking this for a spin on Windows I get different a outcome:

```
(.env36) λ git log -1
commit 7a121d30495ddf4304be6961fdd0a76b57fc5d88 (HEAD -> blueyed/pdb-setup-suspend)
Author: Daniel Hahler <git@thequod.de>
Date:   Thu Apr 11 11:32:09 2019 +0200

    Add test_continue_with_breakpoint_capsys


(.env36) λ pytest testing\test_pdb.py -k with_breakpoint
======================================= test session starts =======================================
platform win32 -- Python 3.6.6, pytest-4.4.1.dev67+g7a121d304, py-1.6.0, pluggy-0.9.0
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('c:\\pytest\\.hypothesis\\examples')
rootdir: c:\pytest, inifile: tox.ini
plugins: hypothesis-3.74.2
collected 55 items / 53 deselected / 2 selected

testing\test_pdb.py ss                                                                       [100%]
===================================== short test summary info =====================================
SKIPPED [2] c:\pytest\src\_pytest\pytester.py:1239: could not import 'pexpect'

============================ 2 skipped, 53 deselected in 0.31 seconds =============================


(.env36) λ pytest .tmp\test-pdb.py
======================================= test session starts =======================================
platform win32 -- Python 3.6.6, pytest-4.4.1.dev67+g7a121d304, py-1.6.0, pluggy-0.9.0
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('c:\\pytest\\.hypothesis\\examples')
rootdir: c:\pytest\.tmp, inifile: pytest.ini
plugins: hypothesis-3.74.2
collected 1 item

.tmp\test-pdb.py
>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> c:\pytest\.tmp\test-pdb.py(8)test_1()
-> f()
(Pdb) break f
Breakpoint 1 at c:\pytest\.tmp\test-pdb.py:1
(Pdb) c

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB continue (IO-capturing resumed) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> c:\pytest\.tmp\test-pdb.py(2)f()
-> print("print_in_f")
(Pdb) c

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB continue (IO-capturing resumed) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
F                                                                           [100%]

============================================ FAILURES =============================================
_____________________________________________ test_1 ______________________________________________

    def test_1():
        __import__('pdb').set_trace()
>       f()

.tmp\test-pdb.py:8:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    def f():
        print("print_in_f")
>       assert 0
E       assert 0

.tmp\test-pdb.py:3: AssertionError
-------------------------------------- Captured stdout call ---------------------------------------
print_in_f
==================================== 1 failed in 4.06 seconds =====================================
Exception ignored in: <async_generator object _ag at 0x00000279844670B0>
Traceback (most recent call last):
  File "C:\Users\Bruno\AppData\Local\Programs\Python\Python36\lib\types.py", line 27, in _ag
  File "C:\Users\Bruno\AppData\Local\Programs\Python\Python36\lib\bdb.py", line 53, in trace_dispatch
  File "C:\Users\Bruno\AppData\Local\Programs\Python\Python36\lib\bdb.py", line 79, in dispatch_call
  File "C:\Users\Bruno\AppData\Local\Programs\Python\Python36\lib\bdb.py", line 176, in break_anywhere
  File "C:\Users\Bruno\AppData\Local\Programs\Python\Python36\lib\bdb.py", line 36, in canonic
AttributeError: 'NoneType' object has no attribute 'abspath'
```

Comment (blueyed) [reply testing/test_pdb.py]: What do you think is different?
Looks like what the test is asserting, no?

```
    child = testdir.spawn_pytest(str(p1))
    child.expect(r"PDB set_trace \(IO-capturing turned off\)")
    child.expect(r"\(Pdb")
    child.sendline("break f")
    child.sendline("c")
    child.expect(r"PDB continue \(IO-capturing resumed\)")
    child.expect(r"PDB set_trace \(IO-capturing turned off\)")
    child.expect(r'-> print\("print_in_f"\)')
    child.expect(r"\(Pdb")
    child.sendline("c")
    rest = child.read().decode("utf8")
    assert "Captured stdout call" in rest
    assert "print_in_f" in rest
    assert "1 failed" in rest
```
(assuming this was `test_continue_with_breakpoint` and not `test_continue_with_breakpoint_capsys`)

Comment (nicoddemus) [reply testing/test_pdb.py]: Argh sorry, brain fart. You can disregard this. But `test_continue_with_breakpoint_capsys` is indeed not showing the expected output in https://github.com/pytest-dev/pytest/pull/5085#discussion_r274670079

File: testing/test_pdb.py (hunk 1 above)
Comment (blueyed) [reply testing/test_pdb.py]: Yeah, here it should mention the fixture ("PDB set_trace \(IO-capturing turned off for fixture capsys\)").
Does it work with capfd instead maybe?

Does `test_pdb_suspends_fixture_capturing` work as expected?

Comment (blueyed) [reply testing/test_pdb.py]: @nicoddemus 
This was hardened in 939362bb.  Can you recheck it, please?

Comment (blueyed) [reply testing/test_pdb.py]: s/hardened/broken/ - looking into it.

**Comment by blueyed** [comment]
Added changelog.
I came back to this because I started to basically fix this again - would be nice to get this merged soon.. :)

**Comment by blueyed** [comment]
Will take a different approach.

**Pull request closed without merging**